### PR TITLE
Add baseline change comparison to the report tables

### PR DESCRIPTION
**Pull request by dupliaka**
Scales gets as a baseline options.
Then each row of the table recalculates due to the baseline 

image

### CODE REVIEW

File: webapp/src/domain/reports/TableReportView.tsx [hunk 1]
```diff
@@ -382,6 +435,32 @@ export default function TableReportView(props: TableReportViewProps) {
                 editable={props.editable}
                 onUpdate={text => update(props.report, comment0, text, 0)}
             />
+
+            <div>
+            <Title headingLevel="h2">Baseline scale</Title>
```
[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:440]: Sorry how can I run those checks?

File: webapp/src/domain/reports/TableReportView.tsx [hunk 2]
```diff
@@ -168,6 +173,23 @@ function ComponentTable(props: ComponentTableProps) {
                                             unit={props.unit}
                                             selector={props.selector}
                                             siblingSelector={props.siblingSelector}
+                                            />
+                                            </Td>
+                                        ))}
+                                    </Tr>
+                                ))}
+                        {scales.slice(1,scales.length).map((sc) => (
```
[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:181]: Oh, that a good one. Thanks a lot!

File: webapp/src/domain/reports/TableReportView.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (rvansa) [reply webapp/src/domain/reports/TableReportView.tsx:440]: Did you run `mvn test` or `mvn package` before the commit? Whenever you run `npm` the git hook should get installed, and then `git commit` would run the linter. You should see something like
```
16:37:30@rvansa horreum$ git commit -m test

> node@0.1.0 prettier-format
> prettier --config .prettierrc "**/*.{ts,tsx}" --write

src/About.tsx 84ms
src/alerts.tsx 75ms
src/App.tsx 48ms
...
```

File: webapp/src/domain/reports/TableReportView.tsx [hunk 3]
```diff
@@ -230,6 +252,31 @@ function ComponentTable(props: ComponentTableProps) {
...
+    const data = parseFloat(props.data.values[0])
+    const baseline = parseFloat(props.baseline.values[0])
+
+    if (isNaN(data) || isNaN(baseline)) {
+        return <>(data error)</>
+    } 
+
+    const change = (data/baseline -1) * 100
+    return (
+            <Button variant="link">
```
[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:274]: Just to keep the style. Any suggestions? What would be the action?

File: webapp/src/domain/reports/TableReportView.tsx [hunk 4]
```diff
@@ -366,6 +413,12 @@ export default function TableReportView(props: TableReportViewProps) {
     const categoryFormatter = formatter(config.categoryFormatter)
     const comment0 = props.report.comments.find(c => c.level === 0)
 
+    const scales = [...new Set(props.report.data.map(d => d.scale))].sort(numericCompare)
+    const scaleFormatter = formatter(config.scaleFormatter)
+
+    const [dropdownOpen, setDropdownOpen] = useState(false)
+    const [baseline, setBaseline] = useState("No Baseline")
```
[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:420]: Do you mean to use ```    const [baseline, setBaseline] = useState()```

File: webapp/src/domain/reports/TableReportView.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:440]: No idea why it didnt work for this commit.
The hook works fine after I have installed prettier ```npm add prettier prettier-eslint prettier-eslint-cli -D```

[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:440]: I think I might applied my ide lint somehow between mvn test and committing the results

[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:440]: Thats a good point to add this to the CONTRIBUTION.md :wink:

File: webapp/src/domain/reports/TableReportView.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (rvansa) [reply webapp/src/domain/reports/TableReportView.tsx:420]: Yes, and compare vs. `undefined` later below.

File: webapp/src/domain/reports/TableReportView.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rvansa) [reply webapp/src/domain/reports/TableReportView.tsx:274]: Oh, I didn't notice I've used that structure as well - okay, nevermind then :) We'll fix it later if desired.

File: webapp/src/domain/reports/TableReportView.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx]: Fixed 8a3a55b7

File: webapp/src/domain/reports/TableReportView.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (rvansa) [reply webapp/src/domain/reports/TableReportView.tsx:422]: I don't see this updated in the last commits, do you prefer to keep this as "No Baseline"?

[the remark this answers is not in the document]
Comment (dupliaka) [reply webapp/src/domain/reports/TableReportView.tsx:422]: Yup I tried to do that, but it then fails on type-resolving at 

Argument of type 'string' is not assignable to parameter of type 'SetStateAction<undefined>'

https://github.com/dupliaka/Horreum/blob/8a3a55b75b011ae87bcb2d3062f0849d56756d5e/webapp/src/domain/reports/TableReportView.tsx#L446

**Comment by rvansa** [comment]
Integrated, thanks!

**Pull request merged**